frontend/src/metabase/visualizations/lib/settings/column.js

```diff
@@ -523,12 +524,22 @@ export const buildTableColumnSettings = ({
     widget: ChartSettingTableColumns,
     getHidden: (series, vizSettings) => vizSettings["table.pivot"],
     isValid: ([{ card, data }]) => {
+      const question = new Question(card /* metadata */);
```

Comment:
the most problematic question, whether we need metadata here or not

Reply:
I think we do. I checked BE code and there was this call chain:

`find-column-indexes-from-legacy-refs` -> `find-column-indexes-for-refs` -> `find-matching-column` -> `resolve-field-id` (uses metadata, will throw if field is not found by id) -> `saved-question-metadata` (uses metadata)

Reply:
I checked those methods as well, but I couldn't say it will 100% throw